src/hardware/voodoo.cpp

```diff
@@ -6996,6 +7054,54 @@ static constexpr uint32_t voodoo_r(const uint32_t addr)
 	return 0xffffffff;
 }
 
+// Get the number of additional threads to use for Voodoo work based on the
+// user's conf setting. By default we use up to 7 threads however the user can
+// customize this.
+
+static int get_num_threads()
+{
+	auto get_auto_threads = []() {
+		// Sweet spot found to be seven threads based on benchmarks:
+		// https://github.com/schellingb/dosbox-pure/issues/300#issuecomment-2144050898
+		constexpr auto MinThreads = 0;
+		constexpr auto MaxThreads = 7;
+		return std::clamp(SDL_GetCPUCount() - 1, MinThreads, MaxThreads);
+	};
+
+	auto parse_positive_base10_int = [](const std::string& str) {
+		constexpr auto Base10 = 10;
+		const auto maybe_int  = parse_int(str, Base10);
+		return (maybe_int && *maybe_int >= 0) ? maybe_int : std::nullopt;
+	};
+
+	// Get the user's "[voodoo] voodoo_threads" setting
+	constexpr auto DefaultSetting = "auto";
+	const auto sec = dynamic_cast<Section_prop*>(control->GetSection("voodoo"));
+	const auto user_setting = sec ? sec->Get_string("voodoo_threads")
+	                              : DefaultSetting;
+
+	// Did the user provide the default or a standardized positive setting?
+	if (user_setting == DefaultSetting || has_true(user_setting)) {
```

Comment:
We allow `auto`, or a number, so `has_true` needs to be ditched. Allowing yes/no, enabled/disabled, etc. confuses the matters for no good reason, so please drop it.

Reply:
It would be slightly more descriptive if you renamed `DefaultSetting` to `AutoSetting`.

Reply:
@FeralChild64 suggested that 'none' be allowed to match other conf settings, but most (all?) numeric settings don't allow 'none'; for example: `[mixer] rate = none` and `[sblaster] irq = none` aren't allowed..  So OK, that makes sense.
Fixed both.